libcloud/dns/drivers/pointdns.py

```diff
@@ -0,0 +1,626 @@
+        :type zone: :class:`Zone`
+        """
+        response = self.connection.request('/zones/%s/mail_redirects' %
+                                           zone.id)
+        mailredirects = self._to_mailredirects(response.object, zone)
+        return mailredirects
+
+    def ex_create_redirect(self, redirect_to, name, type, zone, iframe=None,
+                           query=None):
+        """
+        :param redirect_to: The data field. (redirect_to)
+        :type redirect_to: ``str``
+
+        :param name: The FQDN for the record.
+        :type name: ``str``
+
+        :param type: The type of redirects 301, 302 or 0 for iframes.
+        :type type: ``str``
+
+        :param zone: Zone to list redirects for.
+        :type zone: :class:`Zone`
+
+        :param iframe: Title of iframe (optional).
+        :type iframe: ``str``
+
+        :param query: boolean Information about including query string when
+                      redirecting. (optional).
+        :type query: ``bool``
+        """
+        r_json = {'name': name, 'redirect_to': redirect_to}
+        if type is not None:
+            r_json['redirect_type'] = type
+        if iframe is not None:
+            r_json['iframe_title'] = iframe
+        if query is not None:
+            r_json['redirect_query_string'] = query
+        r_data = json.dumps({'zone_redirect': r_json})
+        response = self.connection.request('/zones/%s/redirects' % zone.id,
+                                           method='POST', data=r_data)
+        redirect = self._to_redirect(response.object, zone.id)
+        return redirect
+
+    def ex_create_mailredirect(self, destination, source, zone):
+        """
+        :param destination: The destination address of mail redirect.
+        :type destination: ``str``
+
+        :param source: The source address of mail redirect.
+        :type source: ``str``
+
+        :param zone: Zone to list redirects for.
+        :type zone: :class:`Zone`
+        """
+        r_json = {'destination_address': destination, 'source_address': source}
+        r_data = json.dumps({'zone_mail_redirect': r_json})
+        response = self.connection.request('/zones/%s/mail_redirects' %
+                                           zone.id, method='POST',
+                                           data=r_data)
+        mailredirect = self._to_mailredirect(response.object, zone.id)
+        return mailredirect
+
+    def ex_get_redirect(self, zone, redirect_id):
+        """
+        :param zone: Zone to list redirects for.
+        :type zone: :class:`Zone`
+
+        :param redirect_id: Redirect id.
+        :type redirect_id: ``str``
+        """
+        response = self.connection.request('/zones/%s/redirects/%s' %
+                                           (zone.id, redirect_id))
+        redirect = self._to_redirect(response.object, zone.id)
+        return redirect
+
+    def ex_get_mailredirects(self, zone, mail_r_id):
```

Comment:
I will make a minor change while merging this pr - `mailredirects` -> `mail_redirects` hope you don't mind :)